### PR TITLE
BUG: resample with ArrowDtype

### LINKED ISSUE #55989: BUG: Resample doesn't work for timestamp[ns][pyarrow]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
from pandas import Timestamp
df = pd.DataFrame.from_dict({Timestamp('2022-03-31 01:00:00'): 1}, orient="index")
df.index = df.index.astype("timestamp[ns][pyarrow]")
df.resample("1min").mean()
```

### Issue Description

Using resample with timestamp[ns][pyarrow] returns a TypeError
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[3], line 5
      3 df = pd.DataFrame.from_dict({Timestamp('2022-03-31 01:00:00'): 1}, orient="index")
      4 df.index = df.index.astype("timestamp[ns][pyarrow]")
----> 5 df.resample("1min").mean()

File ~\AppData\Local\pypoetry\Cache\virtualenvs\\lib\site-packages\pandas\core\generic.py:9439, in NDFrame.resample(self, rule, axis, closed, label, convention, kind, on, level, origin, offset, group_keys)
   9436 else:
   9437     axis = 0
-> 9439 return get_resampler(
   9440     cast("Series | DataFrame", self),
   9441     freq=rule,
   9442     label=label,
   9443     closed=closed,
   9444     axis=axis,
   9445     kind=kind,
   9446     convention=convention,
   9447     key=on,
   9448     level=level,
   9449     origin=origin,
   9450     offset=offset,
   9451     group_keys=group_keys,
   9452 )

File ~\AppData\Local\pypoetry\Cache\virtualenvs\\lib\site-packages\pandas\core\resample.py:1970, in get_resampler(obj, kind, **kwds)
   1966 """
   1967 Create a TimeGrouper and return our resampler.
   1968 """
   1969 tg = TimeGrouper(**kwds)
-> 1970 return tg._get_resampler(obj, kind=kind)

File ~\AppData\Local\pypoetry\Cache\virtualenvs\\lib\site-packages\pandas\core\resample.py:2160, in TimeGrouper._get_resampler(self, obj, kind)
   2151 elif isinstance(ax, TimedeltaIndex):
   2152     return TimedeltaIndexResampler(
   2153         obj,
   2154         timegrouper=self,
   (...)
   2157         gpr_index=ax,
   2158     )
-> 2160 raise TypeError(
   2161     "Only valid with DatetimeIndex, "
   2162     "TimedeltaIndex or PeriodIndex, "
   2163     f"but got an instance of '{type(ax).__name__}'"
   2164 )

TypeError: Only valid with DatetimeIndex, TimedeltaIndex or PeriodIndex, but got an instance of 'Index'
```

### Expected Behavior

Commenting out the conversion to timestamp[ns][pyarrow] produces the expected behaviour. 
```
import pandas as pd
from pandas import Timestamp
df = pd.DataFrame.from_dict({Timestamp('2022-03-31 01:00:00'): 1}, orient="index")
#df.index = df.index.astype("timestamp[ns][pyarrow]")
df.resample("1min").mean()
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit              : 2a953cf80b77e4348bf50ed724f8abc0d814d9dd
python              : 3.10.4.final.0
python-bits         : 64
OS                  : Windows
OS-release          : 10
Version             : 10.0.22621
machine             : AMD64
processor           : Intel64 Family 6 Model 165 Stepping 2, GenuineIntel
byteorder           : little
LC_ALL              : None
LANG                : None
LOCALE              : English_United Kingdom.1252

pandas              : 2.1.3
numpy               : 1.24.2
pytz                : 2023.3
dateutil            : 2.8.2
setuptools          : 67.6.1
pip                 : 23.2
Cython              : 3.0.2
pytest              : 7.3.0
hypothesis          : None
sphinx              : None
blosc               : None
feather             : None
xlsxwriter          : None
lxml.etree          : None
html5lib            : None
pymysql             : None
psycopg2            : None
jinja2              : 3.1.2
IPython             : 8.12.0
pandas_datareader   : None
bs4                 : 4.12.2
bottleneck          : None
dataframe-api-compat: None
fastparquet         : None
fsspec              : 2023.10.0
gcsfs               : None
matplotlib          : 3.7.3
numba               : 0.57.1
numexpr             : None
odfpy               : None
openpyxl            : None
pandas_gbq          : None
pyarrow             : 13.0.0
pyreadstat          : None
pyxlsb              : None
s3fs                : None
scipy               : 1.10.1
sqlalchemy          : 2.0.23
tables              : None
tabulate            : 0.9.0
xarray              : 2023.8.0
xlrd                : None
zstandard           : None
tzdata              : 2023.3
qtpy                : None
pyqt5               : None

</details>

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #55989 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

The current resample path heavily uses `DatetimeIndex`/`TimedeltaIndex` APIs, so it's kinda onerous to try to allow `Index` API methods. This essentially wraps the Arrow types into one of the two classes above and saves an `_arrow_dtype` for the output

### CODE REVIEW

File: pandas/core/resample.py [hunk 1]
```diff
@@ -2495,6 +2501,17 @@ def _get_period_bins(self, ax: PeriodIndex):
 
         return binner, bins, labels
 
+    def _set_grouper(
+        self, obj: NDFrameT, sort: bool = False, *, gpr_index: Index | None = None
+    ) -> tuple[NDFrameT, Index, npt.NDArray[np.intp] | None]:
+        obj, ax, indexer = super()._set_grouper(obj, sort, gpr_index=gpr_index)
+        if isinstance(ax.dtype, ArrowDtype) and ax.dtype.kind in "Mm":
```
Comment (rhshadrach) [inline pandas/core/resample.py:2508]: We don't want to do this in `Grouper`'s implementation, is that right?

File: pandas/tests/resample/test_datetime_index.py [hunk 2]
```diff
@@ -2195,3 +2197,26 @@ def test_resample_b_55282(unit):
...
+            "UTC",
+            marks=pytest.mark.xfail(
+                condition=is_platform_windows(),
+                reason="TODO: Set ARROW_TIMEZONE_DATABASE env var in CI",
+            ),
+        ),
+    ],
+)
+def test_arrow_timestamp_resample(tz):
+    idx = Series(date_range("2020-01-01", periods=5), dtype="timestamp[ns][pyarrow]")
```
Comment (rhshadrach) [inline pandas/tests/resample/test_datetime_index.py:2217]: GH#

File: pandas/tests/resample/test_timedelta.py [hunk 3]
```diff
@@ -207,3 +209,11 @@ def test_resample_closed_right():
         ),
     )
     tm.assert_series_equal(result, expected)
+
+
+@td.skip_if_no("pyarrow")
+def test_arrow_duration_resample():
+    idx = pd.Index(timedelta_range("1 day", periods=5), dtype="duration[ns][pyarrow]")
```
Comment (rhshadrach) [inline pandas/tests/resample/test_timedelta.py:216]: GH#

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/core/resample.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/core/resample.py:2508]: I only see `_set_grouper` being used in one place outside of resample, and where it is the `ax` return is ignored. So I _think_we could just add this to `Grouper`'s implementation, which reduces a little bit of the complexity, but at the cost of doing extra work that just gets discarded (outside of resample).

Definitely good with this as-is.

**Review by rhshadrach — APPROVED** [review]
lgtm

File: pandas/core/resample.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/resample.py:2508]: Ah I see. I'll revisit this once the Grouper attribute deprecation are enforced. I think I can simplify this future once those are enforced.

**Pull request merged**